### PR TITLE
[MacOS][NativeWindowing] Programmatically move windows when user changes monitor setting

### PR DESCRIPTION
**Pull request by enen92**
## Description
This continues the endeavour to improve macOS native windowing and its UX by bringing a few improvements also for settings and windowing systems in general.

When a user changes the monitor setting manually, the window should be moved to the selected screen. MacOS doesn't allow moving windows when running on fullscreen so, a new `SupportsScreenMove` complex condition was added to settings that ends up calling the respective windowing implementation for verification. With this, I've found another bug where the visibility (and/or enable status) of settings that depend on complex conditions was not being updated

Hence, this PR:
- Disables the monitor setting if running fullscreen on macOS (nativewindowing)
- Moves kodi to the proper screen when the user changes the setting in windowed mode

**Windowed (window will be moved if another screen is selected):**
image

**Fullscreen (setting disabled)**
image

## Motivation and context
Improve UX for native windowing on macos

## How has this been tested?
Runtime tested on macos (apple silicon) with native windowing

## What is the effect on users?
Better immediate feedback to settings change

### CODE REVIEW

**Comment by enen92** [comment]
Merging by the end of the weekend unless someone opposes.

File: xbmc/windowing/WinSystem.h [hunk 1]
```diff
@@ -109,6 +109,12 @@ class CWinSystemBase
   // notifications
   virtual void OnMove(int x, int y) {}
 
+  /**
+   * \brief Used to signal the windowing system about the intention of the user to change the main display
+   * \details triggered, for example, when the user manually changes de monitor setting
```
Comment (kambala-decapitator) [inline xbmc/windowing/WinSystem.h:114]:
```suggestion
   * \details triggered, for example, when the user manually changes the monitor setting
```

File: xbmc/windowing/osx/WinSystemOSX.h [hunk 2]
```diff
@@ -59,6 +59,17 @@ class CWinSystemOSX : public CWinSystemBase, public ITimerCallback
 
   std::string GetClipboardText() override;
 
+  /*! \brief Check if the windowing system supports moving windows across screens
+   *  \return true if the windowing system supports moving windows across screens, false otherwise
+  */
```
Comment (kambala-decapitator) [inline xbmc/windowing/osx/WinSystemOSX.h:64]: not sure that copy-pasting docstring from parent class is needed

File: xbmc/windowing/osx/WinSystemOSX.h [hunk 3]
```diff
@@ -59,6 +59,17 @@ class CWinSystemOSX : public CWinSystemBase, public ITimerCallback
...
+  bool SupportsScreenMove() override;
+
+  /**
+   * \brief Used to signal the windowing system about the intention of the user to change the main display
+   * \details triggered, for example, when the user manually changes de monitor setting
+  */
```
Comment (kambala-decapitator) [inline xbmc/windowing/osx/WinSystemOSX.h:70]: not sure that copy-pasting docstring from parent class is needed

**Review by kambala-decapitator — APPROVED** [review]
> MacOS doesn't allow moving windows when running on fullscreen

do you think it'd work if you go out of fullscreen programmatically and then move the window to another screen?

File: xbmc/windowing/osx/WinSystemOSX.h (hunk 2 above)
Comment (enen92) [reply xbmc/windowing/osx/WinSystemOSX.h:64]: doesn't harm either :)

**Comment by enen92** [comment]
> do you think it'd work if you go out of fullscreen programmatically and then move the window to another screen?

It's a bit hacky and also not trivial since the toggle operations are asynchronous and we react to a lot of events.
Might try to have an attempt at it later again but for now I think it is good enough to simply disable the setting. This PR also brings a couple of other improvements for settings apart from this particular change

**Pull request merged**